### PR TITLE
root-cas setting

### PR DESCRIPTION
**Pull request by jrobsonchase**
closes: #369 

## What
Utilizes the `trusted` or `host` CA as the source of truth via a helm install flag.

## How
Takes an install option for `--set rootCAs=host`  and plumb the `isHostCA` check into the `caCerts` for it to just get the host certs. 

## Verification

- Build 2048 image, copying in cert and updating certs, apply changes
- Set `--set rootCAs host` and make deploy, no controller errors ✅ and can play example
- Set `--set rootCAs host` with canonical 2048 example without cert gives errors in controller
- Set `--set rootCAs trusted` no errors and plays 2048 as expected
- no set plays as expected

### CODE REVIEW

**Comment by jrobsonchase** [comment]
Looks to be working as expected:

```
failed to reconnect session     {"obj": "csess", "id": "0eb6ded7abdf", "err": "failed to send authentication request: tls: failed to verify certificate: x509: certificate signed by unknown authority"}
```

with the standard agent ingress, which uses the internal ngrok CA rather than Let's Encrypt.

**Comment by bobzilladev** [comment]
From meeting today:

1. Will instead add a `root_cas` option with similar semantics to the agent: https://ngrok.com/docs/ngrok-agent/config/#root_cas
2. The default will be `internal` and use the cert baked into ngrok-go. Caveat: if the special local directories exist and have certs, those will be used instead (same behavior as the controller currently uses).
3. The `host` option will use host certs, as this PR makes possible.
4. Any other values for `root_cas` will throw an error. In the future we may build functionality to refer to a k8s object here.

A table @jrobsonchase made of options, for posterity. We're going with the middle option:
2024-05-17-105728_691x276_scrot

File: cmd/main.go [hunk 1]
```diff
@@ -122,7 +122,7 @@ func cmd() *cobra.Command {
 	c.Flags().StringVar(&opts.watchNamespace, "watch-namespace", "", "Namespace to watch for Kubernetes resources. Defaults to all namespaces.")
 	c.Flags().StringVar(&opts.managerName, "manager-name", "ngrok-ingress-controller-manager", "Manager name to identify unique ngrok ingress controller instances")
 	c.Flags().BoolVar(&opts.useExperimentalGatewayAPI, "use-experimental-gateway-api", false, "sets up experemental gatewayAPI")
-	c.Flags().BoolVar(&opts.hostCA, "host-ca", false, "use the host's default CA store for the ngrok agent connection")
+	c.Flags().StringVar(&opts.rootCAs, "root-cas", "internal", "internal (default) or host: use the internalt ngrok agent CA or the host CA")
```
[the remark this answers is not in the document]
Comment (OfTheDelmer) [reply cmd/main.go:125]: Oops, I went too literal with the `internal` naming from the matrix above, but `trusted` would be the consistent value.

File: pkg/tunneldriver/driver.go [hunk 2]
```diff
@@ -97,15 +100,28 @@ func New(ctx context.Context, logger logr.Logger, opts TunnelDriverOpts, tunnelC
 		connOpts = append(connOpts, ngrok.WithServer(opts.ServerAddr))
 	}
 
-	// Only configure custom certs if the directory exists
-	if _, err := os.Stat(customCertsPath); !os.IsNotExist(err) {
-		caCerts, err := caCerts()
+	isHostCA := opts.RootCAs == "host"
+
+	// validate is default "internal" or "host
+	if !isHostCA && opts.RootCAs != "internal" {
```
[the remark this answers is not in the document]
Comment (OfTheDelmer) [reply pkg/tunneldriver/driver.go:106]: I can allow empty string here. In main it will default to `internal` if unset.

**Review by bobzilladev — APPROVED** [review]

**Pull request merged**